### PR TITLE
Add pod spec resources limits & requests to all prow jobs

### PR DESCRIPTION
**Pull request by jljaco**
As a followup to the [recent switchover to Karpenter](https://github.com/aws-controllers-k8s/test-infra/pull/308) in our Prow cluster, this change adds `resources.limits` and `resources.requests` to all Prow job templates.

Using:
- 1 VCPU 
- 500Mi of memory 

as a starting point for requests, with the intent to tune going forward as needed.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by A-Hilaly — COMMENTED** [review]
Did you mean `Mi` instead of `Gi`?

**Comment by jljaco** [comment]
> Did you mean `Mi` instead of `Gi`?

Oh wow, yes nice catch! ⚾

**Comment by jljaco** [comment]
> Did you mean `Mi` instead of `Gi`?

Fixed!

File: prow/jobs/jinja/periodics/label_sync.jinja2 [hunk 1]
```diff
@@ -11,6 +11,13 @@
     containers:
     - name: label-sync
       image: gcr.io/k8s-prow/label_sync:v20221205-a1b0b85d88
+      resources:
+        limits:
+          cpu: 1
+        requests:
+          cpu: 1
+          memory: "500Mi"
+          ephemeral-storage: "100Gi"
```
Comment (RedbackThomson) [inline prow/jobs/jinja/periodics/label_sync.jinja2:20]: This is not something we want to request. In the system that was running for 2 years, we had 100Gi total for the entire node. There is no reason jobs should have access to 100Gi of local storage. I think we should just remove the `ephemeral-storage` request altogether.

Comment (jljaco) [reply prow/jobs/jinja/periodics/label_sync.jinja2]: Removed `ephemeral-storage` from all requests

File: prow/jobs/jinja/periodics/label_sync.jinja2 [hunk 2]
```diff
@@ -11,6 +11,12 @@
     containers:
     - name: label-sync
       image: gcr.io/k8s-prow/label_sync:v20221205-a1b0b85d88
+      resources:
+        limits:
+          cpu: 2
```
Comment (jaypipes) [inline prow/jobs/jinja/periodics/label_sync.jinja2:16]: Is there any reason the label-sync job needed 2 CPUs? :)

File: prow/jobs/jinja/presubmits/runtime_tests.jinja2 [hunk 3]
```diff
@@ -9,6 +9,9 @@
       - image: {{ image_context.images["unit-test"] }}
         resources:
           limits:
+            cpu: 4
```
Comment (jaypipes) [inline prow/jobs/jinja/presubmits/runtime_tests.jinja2:12]: Pretty sure runtime unit tests don't need >1 CPU :)

File: prow/jobs/jinja/presubmits/service_tests.jinja2 [hunk 4]
```diff
@@ -94,7 +106,10 @@
       - image: {{ image_context.images["unit-test"] }}
         resources:
           limits:
-            cpu: 2
+            cpu: 4
```
Comment (jaypipes) [inline prow/jobs/jinja/presubmits/service_tests.jinja2:109]: Unit tests don't need >1 CPU either

**Pull request merged**